openfermioncirq/gates/common_gates.py

```diff
@@ -12,45 +12,66 @@
 
 """Gates that are commonly used for quantum simulation of fermions."""
 
-from typing import Optional, Union
+from typing import Optional, Union, Tuple
 
 import numpy
 
 import cirq
 
 
-class FermionicSwapGate(cirq.InterchangeableQubitsGate,
-                        cirq.KnownMatrix,
-                        cirq.ReversibleEffect,
-                        cirq.TextDiagrammable,
+class FermionicSwapGate(cirq.EigenGate,
+                        cirq.InterchangeableQubitsGate,
                         cirq.TwoQubitGate):
     """Swaps two adjacent fermionic modes under the JWT."""
 
-    def matrix(self):
-        return numpy.array([[1, 0, 0, 0],
-                            [0, 0, 1, 0],
-                            [0, 1, 0, 0],
-                            [0, 0, 0, -1]])
+    def __init__(self, *,  # Forces keyword args.
+                 half_turns: Union[cirq.Symbol, float] = 1.0) -> None:
+        super().__init__(exponent=half_turns)
+
+    def _eigen_components(self):
+        return [
+            (0, numpy.array([[1, 0,   0,   0],
+                             [0, 0.5, 0.5, 0],
+                             [0, 0.5, 0.5, 0],
+                             [0, 0,   0,   0]])),
+            (1, numpy.array([[0,  0,    0,   0],
+                             [0,  0.5, -0.5, 0],
+                             [0, -0.5,  0.5, 0],
+                             [0,  0,    0,   1]])),
+        ]
 
-    def inverse(self):
-        return self
+    def _canonical_exponent_period(self) -> Optional[float]:
+        return 2
 
-    def text_diagram_info(self, args: cirq.TextDiagramInfoArgs
-                          ) -> cirq.TextDiagramInfo:
+    def _with_exponent(self,
+                       exponent: Union[cirq.Symbol, float]
+                       ) -> 'FermionicSwapGate':
+        return FermionicSwapGate(half_turns=exponent)
+
+    @property
+    def half_turns(self) -> Union[cirq.Symbol, float]:
+        return self._exponent
+
+    def _circuit_diagram_info_(self, args: cirq.CircuitDiagramInfoArgs
+                               ) -> Tuple[str, str]:
         if args.use_unicode_characters:
-            wire_symbols = ('×ᶠ', '×ᶠ')
-        else:
-            wire_symbols = ('fswap', 'fswap')
-        return cirq.TextDiagramInfo(wire_symbols=wire_symbols)
+            return '×ᶠ', '×ᶠ'
+        return 'fswap', 'fswap'
 
-    def __repr__(self):
-        return 'FSWAP'
+    def __str__(self) -> str:
+        if self.half_turns == 1:
+            return 'FSWAP'
+        return 'FSWAP**{!r}'.format(self.half_turns)
+
+    def __repr__(self) -> str:
+        if self.half_turns == 1:
+            return 'FSWAP'
```

Comment:
`openfermioncirq.FSWAP` or `ofc.FSWAP`

Checkout `cirq.assert_equivalent_repr`. You should define an equivalent for openfermion with the appropriate imports as the default setup code.